frontend/test/pytest/test_template.py

```diff
@@ -683,8 +1180,299 @@ def qnn(weights, inputs):
+    x = 5
+    y = 6
+    mod = 7
+
+    x_wires = [0, 1, 2]
+    y_wires = [3, 4, 5]
+    output_wires = [7, 8, 9]
+    work_wires = [6, 10]
+
+    def out_adder():
+        qml.Identity(0)
+        qml.BasisEmbedding(x, wires=x_wires)
+        qml.BasisEmbedding(y, wires=y_wires)
+        qml.OutAdder(x_wires, y_wires, output_wires, mod, work_wires)
+        return qml.sample(wires=output_wires)
+
+    device = qml.device(backend, wires=11, shots=10000)
+    interpreted_fn = qml.QNode(out_adder, device)
+    jitted_fn = qjit(interpreted_fn)
+
+    interpreted_result = interpreted_fn()
+    jitted_result = jitted_fn()
+
+    assert np.allclose(interpreted_result, jitted_result)
+
+
+def test_out_multiplier(backend):
+    """Test OutMultiplier."""
+    x = 2
+    y = 7
+    mod = 12
+
+    x_wires = [0, 1]
+    y_wires = [2, 3, 4]
+    output_wires = [6, 7, 8, 9]
+    work_wires = [5, 10]
+
+    def out_multiplier():
+        qml.Identity(0)
+        qml.BasisEmbedding(x, wires=x_wires)
+        qml.BasisEmbedding(y, wires=y_wires)
+        qml.OutMultiplier(x_wires, y_wires, output_wires, mod, work_wires)
+        return qml.sample(wires=output_wires)
+
+    device = qml.device(backend, wires=11, shots=2)
+    interpreted_fn = qml.QNode(out_multiplier, device)
+    jitted_fn = qjit(interpreted_fn)
+
+    interpreted_result = interpreted_fn()
+    jitted_result = jitted_fn()
+
+    assert np.allclose(interpreted_result, jitted_result)
+
+
+def test_phase_adder(backend):
+    """Test PhaseAdder."""
+    x = 8
+    k = 5
+    mod = 15
+
+    x_wires = [0, 1, 2, 3]
+    work_wire = [4]
+
+    def phase_adder():
+        qml.BasisEmbedding(x, wires=x_wires)
+        qml.QFT(wires=x_wires)
+        qml.PhaseAdder(k, x_wires, mod, work_wire)
+        qml.adjoint(qml.QFT)(wires=x_wires)
+        return qml.sample(wires=x_wires)
+
+    device = qml.device(backend, wires=range(5), shots=2)
+    interpreted_fn = qml.QNode(phase_adder, device)
+    jitted_fn = qjit(interpreted_fn)
+
+    interpreted_result = interpreted_fn()
+    jitted_result = jitted_fn()
+
+    assert np.allclose(interpreted_result, jitted_result)
+
+
+@pytest.mark.xfail(reason="Qutrit operators not supported on lightning.")
+def test_qutrit_basis_state_preparation(backend):
+    """Test QutritBasisStatePreparation."""
+    basis_state = [0, 1]
+    wires = [0, 1]
+    obs = np.array([[1, 0, 0], [0, 2, 0], [0, 0, 3]])
+
+    def qutrit_basis_state_preparation(state, obs):
+        qml.QutritBasisStatePreparation(state, wires)
+
+        return [qml.expval(qml.THermitian(A=obs, wires=i)) for i in range(3)]
+
+    device = qml.device(backend, wires=2)
+    interpreted_fn = qml.QNode(qutrit_basis_state_preparation, device)
+    jitted_fn = qjit(interpreted_fn)
+
+    interpreted_result = interpreted_fn(basis_state, obs)
+    jitted_result = jitted_fn(basis_state, obs)
+
+    assert np.allclose(interpreted_result, jitted_result)
+
+
+def test_cosine_window(backend):
```

Comment:
I remember seeing you also had a cosine window PR. Since this test is in here, does that PR still need to be merged/reviewed? Or did that PR have some additional purpose as well?

Reply:
The cosine window PR contains a fix necessary to make this test pass.